prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -412,6 +425,26 @@ class FHIRRouter(
                 }
                 )
 
+            // TODO: merge with condition filter (see https://github.com/CDCgov/prime-reportstream/issues/12705)
+            // MAPPED CONDITION FILTER
+            //  default: allowAll
+            if (bundle.getObservations().isNotEmpty() && receiver.mappedConditionFilter.isNotEmpty()) {
+                val codes = receiver.mappedConditionFilter.codes()
+                val filteredObservations = bundle.getObservationsWithCondition(codes)
+                if (filteredObservations.isEmpty()) {
```

Comment:
This step is checking if the bundle as a whole should get sent. It will not remove any resources from the bundle, just not add the receiver if all observations will get filtered.

I've added a missed check to make this a noop on receivers that have no mapped condition filter configured.

Reply:
I have added logging for pruned observations.